src/cartservice/src/Startup.cs

```diff
@@ -26,23 +27,20 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             string redisAddress = Configuration["REDIS_ADDR"];
-            ICartStore cartStore = null;
             if (!string.IsNullOrEmpty(redisAddress))
             {
-                cartStore = new RedisCartStore(redisAddress);
+                services.AddStackExchangeRedisCache(options =>
+                {
+                    options.Configuration = redisAddress;
+                });
             }
             else
             {
-                Console.WriteLine("Redis cache host(hostname+port) was not specified. Starting a cart service using local store");
-                Console.WriteLine("If you wanted to use Redis Cache as a backup store, you should provide its address via command line or REDIS_ADDR environment variable.");
-                cartStore = new LocalCartStore();
+                Console.WriteLine("Redis cache host(hostname+port) was not specified. Starting a cart service using in memory store");
+                services.AddDistributedMemoryCache();
             }
 
-            // Initialize the redis store
-            cartStore.InitializeAsync().GetAwaiter().GetResult();
-            Console.WriteLine("Initialization completed");
-
-            services.AddSingleton<ICartStore>(cartStore);
+            services.AddSingleton<ICartStore, RedisCartStore>();
```

Comment:
We don't instantiate any `RedisCartStore` object during the initialization of the app. To respect the Dependency Injection pattern, we just define it as a `Singleton` which could be used by anyone later. It's lighter at startup time, avoid performance and race conditions issues too.

Reply:
good catch, makes sense